modules/sfp_phishstats.py

```diff
@@ -0,0 +1,186 @@
+# Name:         sfp_phishstats
+# Purpose:      Spiderfoot plugin to search PhishStats API
+#               to determine if an IP is malicious 
+#
+# Author:      Krishnasis Mandal <krishnasis@hotmail.com>
+#
+# Created:     18/05/2020
+# Copyright:   (c) Steve Micallef
+# Licence:     GPL
+# -------------------------------------------------------------------------------
+
+from sflib import SpiderFoot, SpiderFootPlugin, SpiderFootEvent
+from netaddr import IPNetwork
+import urllib.request, urllib.parse, urllib.error
+import json
+
+class sfp_phishstats(SpiderFootPlugin):
+    """PhishStats:Investigate,Passive:Reputation Systems::Determine if an IP Address is malicious"""
+
+    opts = {
+        'checkaffiliates': True,
+        'subnetlookup': False,
+        'netblocklookup': True,
+        'maxnetblock': 24,
+        'maxsubnet': 24
+    }
+
+    # Option descriptions. Delete any options not applicable to this module.
+    optdescs = {
+        'checkaffiliates': "Check affiliates?",
+        'subnetlookup': "Look up all IPs on subnets which your target is a part of?",
+        'netblocklookup': "Look up all IPs on netblocks deemed to be owned by your target for possible blacklisted hosts on the same target subdomain/domain?",
+        'maxnetblock': "If looking up owned netblocks, the maximum netblock size to look up all IPs within (CIDR value, 24 = /24, 16 = /16, etc.)",
+        'maxsubnet': "If looking up subnets, the maximum subnet size to look up all the IPs within (CIDR value, 24 = /24, 16 = /16, etc.)"
+    }
+
+    results = None
+    errorState = False  
+
+    def setup(self, sfc, userOpts=dict()):
+        self.sf = sfc
+        self.results = self.tempStorage()
+
+        for opt in list(userOpts.keys()):
+            self.opts[opt] = userOpts[opt]
+        
+
+    # What events is this module interested in for input
+    # For a list of all events, check sfdb.py.
+    def watchedEvents(self):
+        return ["IP_ADDRESS", "NETBLOCK_OWNER", "NETBLOCK_MEMBER",
+            "AFFILIATE_IPADDR"]
+
+    # What events this module produces
+    def producedEvents(self):
+        return ["IP_ADDRESS", "MALICIOUS_IPADDR", "RAW_RIR_DATA"]
+
+    # Check whether the IP Address is malicious using Phishstats API
+    # https://phishstats.info/
+    def queryIPAddress(self, qry):
+        params = {
+            '_where': "(ip,eq," + qry.encode('raw_unicode_escape').decode("ascii", errors='replace') + ")",
+            '_size': 1
+        }
+
+        headers = {
+            'Accept' : "application/json",
+        }
+
+        res = self.sf.fetchUrl(
+          'https://phishstats.info:2096/api/phishing?' + urllib.parse.urlencode(params),
+          headers=headers,
+          timeout=15,
+          useragent=self.opts['_useragent']
+        )
+
+        if not res['code'] == "200":
+            self.sf.debug("No information found from Phishstats for IP Address")
```

Comment:
Is there the possibility of another code being returned which doesn't indicate that data wasn't found, but instead indicates a different problem? e.g. `500` (server error) or `429` (throttling)?